test/api/v3/unit/middlewares/cronMiddleware.js

```diff
@@ -0,0 +1,170 @@
+        expect(err).to.not.exist;
+        expect(taskFound).to.exist;
+        done();
+      });
+    });
+  });
+
+  it('should clear todos older than 90 days for subscribed users', (done) => {
+    res.locals.user.purchased.plan.customerId = 'subscribedId';
+    res.locals.user.purchased.plan.dateUpdated = moment('012013', 'MMYYYY');
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+
+    let task = generateTodo(res.locals.user);
+    task.dateCompleted = moment(new Date()).subtract({days: 91});
+    task.completed = true;
+    task.save();
+
+    cronMiddleware(req, res, function callback () {
+      Tasks.Task.findOne({_id: task}, function (err, taskFound) {
+        expect(err).to.not.exist;
+        expect(taskFound).to.not.exist;
+        done();
+      });
+    });
+  });
+
+  it('should call next is user was not modified after cron', (done) => {
+    let hpBefore = res.locals.user.stats.hp;
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+    generateDaily(res.locals.user);
+
+    cronMiddleware(req, res, function callback () {
+      expect(res.locals.user.stats.hp).to.be.equal(hpBefore);
+      done();
+    });
+  });
+
+  it('does damage for missing dailies', (done) => {
+    let hpBefore = res.locals.user.stats.hp;
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+    let daily = generateDaily(res.locals.user);
+    daily.startDate = moment(new Date()).subtract({days: 2});
+    daily.save();
+
+    cronMiddleware(req, res, function callback () {
+      expect(res.locals.user.stats.hp).to.be.lessThan(hpBefore);
+      done();
+    });
+  });
+
+  it('updates tasks', (done) => {
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+    let todo = generateTodo(res.locals.user);
+    let todoValueBefore = todo.value;
+
+    cronMiddleware(req, res, function callback () {
+      Tasks.Task.findOne({_id: todo._id}, function (err, todoFound) {
+        expect(err).to.not.exist;
+        expect(todoFound.value).to.be.lessThan(todoValueBefore);
+        done();
+      });
+    });
+  });
+
+  it('applies quest progress', (done) => {
+    let hpBefore = res.locals.user.stats.hp;
+    res.locals.user.lastCron = moment(new Date()).subtract({days: 2});
+    let daily = generateDaily(res.locals.user);
+    daily.startDate = moment(new Date()).subtract({days: 2});
+    daily.save();
+
+    let questKey = 'dilatory';
+    res.locals.user.party.quest.key = questKey;
+
+    let party = new Group({
+      type: 'party',
+      name: generateUUID(),
+      leader: res.locals.user._id,
+    });
+    party.quest.members[res.locals.user._id] = true;
+    party.quest.key = questKey;
+    party.save();
+
+    res.locals.user.party._id = party._id;
+    res.locals.user.save();
+
+    party.startQuest(res.locals.user);
+
+    cronMiddleware(req, res, function callback () {
+      expect(res.locals.user.stats.hp).to.be.lessThan(hpBefore);
```

Comment:
this does not actually assert on anything related to the quest. How do we know that this was boss damage on not regular daily damage?